nexus/src/external_api/http_entrypoints.rs

```diff
@@ -3380,14 +3392,169 @@ async fn sshkeys_delete_key(
+#[endpoint {
+    method = GET,
+    path = "/by_id/disks/{id}",
+    tags = ["disks"],
+}]
+async fn by_id_disk_get(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<ByIdPathParams>,
+) -> Result<HttpResponseOk<Disk>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let id = &path.id;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let disk = nexus.disk_fetch_by_id(&opctx, id).await?;
+        Ok(HttpResponseOk(disk.into()))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Fetch an image by id
+#[endpoint {
+    method = GET,
+    path = "/by_id/images/{id}",
+    tags = ["images"],
+}]
+async fn by_id_image_get(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<ByIdPathParams>,
+) -> Result<HttpResponseOk<Image>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let id = &path.id;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let image = nexus.image_fetch_by_id(&opctx, id).await?;
+        Ok(HttpResponseOk(image.into()))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Fetch a global image by id
+#[endpoint {
+    method = GET,
+    path = "/by_id/global-images/{id}",
+    tags = ["images:global"],
+}]
+async fn by_id_global_image_get(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<ByIdPathParams>,
+) -> Result<HttpResponseOk<GlobalImage>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let id = &path.id;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let image = nexus.global_image_fetch_by_id(&opctx, id).await?;
+        Ok(HttpResponseOk(image.into()))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Fetch a snapshot by id
+#[endpoint {
+    method = GET,
+    path = "/by_id/snapshots/{id}",
+    tags = ["snapshots"],
+}]
+async fn by_id_snapshot_get(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<ByIdPathParams>,
+) -> Result<HttpResponseOk<Snapshot>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let id = &path.id;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let snapshot = nexus.snapshot_fetch_by_id(&opctx, id).await?;
+        Ok(HttpResponseOk(snapshot.into()))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Fetch a network interface by id
+#[endpoint {
+    method = GET,
+    path = "/by_id/network-
```

Comment:
hahaha. You can see where my computer ran out of memory and stopped writing to the file. 🤦‍♂️